src/DocumentFragment.php

```diff
@@ -21,4 +21,50 @@
 class DocumentFragment extends \DOMDocumentFragment {
 use LiveProperty, ParentNode;
 
+/**
+ * @param string $id
+ * @return Element|null
+ */
+public function getElementById(string $id)/*:?Element*/ {
+	return $this->callMethodOnTemporaryElement(
+		"querySelector", ["#" . $id]);
+}
+
+/**
+ * @param string $selectors CSS selector(s)
+ * @return Element|null
+ */
+public function querySelector(string $selectors) {
+    return $this->callMethodOnTemporaryElement(
+        "querySelector", [$selectors]);
+}
+
+/**
+ * @param string $selectors CSS selector(s)
+ * @return HTMLCollection
+ */
+public function querySelectorAll(string $selectors):HTMLCollection {
+    return $this->callMethodOnTemporaryElement(
+        "querySelectorAll", [$selectors]);
+}
+
+private function callMethodOnTemporaryElement(string $methodName, array $args) {
+	$childNodes = [];
+
+    $tempElement = $this->ownerDocument->createElement("document-fragment");
```

Comment:
Not sure ... I've started using PHP Storm so it could potentially be my misunderstanding of the new editor.